datasets/opus_paracrawl/opus_paracrawl.py

```diff
@@ -0,0 +1,132 @@
+# coding=utf-8
+# Copyright 2020 HuggingFace Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+import os
+
+import datasets
+
+
+_DESCRIPTION = """\
+Parallel corpora from Web Crawls collected in the ParaCrawl project
+40 languages, 41 bitexts
+total number of files: 20,995
+total number of tokens: 21.40G
+total number of sentence fragments: 1.12G
+"""
+_HOMEPAGE_URL = "http://opus.nlpl.eu/ParaCrawl.php"
+_CITATION = """\
+@InProceedings{TIEDEMANN12.463,
+  author = {J{\"o}rg Tiedemann},
+  title = {Parallel Data, Tools and Interfaces in OPUS},
+  booktitle = {Proceedings of the Eight International Conference on Language Resources and Evaluation (LREC'12)},
+  year = {2012},
+  month = {may},
+  date = {23-25},
+  address = {Istanbul, Turkey},
+  editor = {Nicoletta Calzolari (Conference Chair) and Khalid Choukri and Thierry Declerck and Mehmet Ugur Dogan and Bente Maegaard and Joseph Mariani and Jan Odijk and Stelios Piperidis},
+  publisher = {European Language Resources Association (ELRA)},
+  isbn = {978-2-9517408-7-7},
+  language = {english}
+ }
+"""
+
+_VERSION = "1.0.0"
+_BASE_NAME = "ParaCrawl.{}.{}"
+_BASE_URL = "https://object.pouta.csc.fi/OPUS-ParaCrawl/v7.1/moses/{}-{}.txt.zip"
+# Please note that only few pairs are shown here. You can use config to generate data for all language pairs
+_LANGUAGE_PAIRS = [
+    ("el", "en"),
+    ("en", "ha"),
+    ("en", "ig"),
+    ("en", "km"),
+    ("en", "so"),
+    ("de", "pl"),
+    ("fr", "nl"),
+    ("en", "sw"),
+    ("en", "tl"),
+    ("es", "gl"),
+]
+
+
+class ParaCrawlConfig(datasets.BuilderConfig):
+    def __init__(self, *args, lang1=None, lang2=None, **kwargs):
+        super().__init__(
+            *args,
+            name=f"{lang1}-{lang2}",
+            **kwargs,
+        )
+        self.lang1 = lang1
+        self.lang2 = lang2
```

Comment:
It would be nice to have an `assert` here to ensure:
1. The languages are supported (ie present in the dataset)
2. The languages are different